cli/src/main/java/org/openjdk/skara/cli/GitPr.java

```diff
@@ -729,31 +742,51 @@ public static void main(String[] args) throws IOException, InterruptedException
+            appendPaddedHTMLComment(file, "");
+            appendPaddedHTMLComment(file, "Commits to be included from branch '" + currentBranch.name() + "':");
             for (var commit : commits) {
                 var desc = commit.hash().abbreviate() + ": " + commit.message().get(0);
-                Files.writeString(file, "# - " + desc + "\n", StandardOpenOption.APPEND);
+                appendPaddedHTMLComment(file, "- " + desc);
+                if (!commit.isMerge()) {
+                    var diff = commit.parentDiffs().get(0);
+                    for (var patch : diff.patches()) {
+                        var status = patch.status();
+                        if (status.isModified()) {
+                            appendPaddedHTMLComment(file, "  M  " + patch.target().path().get().toString());
+                        } else if (status.isAdded()) {
+                            appendPaddedHTMLComment(file, "  A  " + patch.target().path().get().toString());
+                        } else if (status.isDeleted()) {
+                            appendPaddedHTMLComment(file, "  D  " + patch.source().path().get().toString());
+                        } else if (status.isRenamed()) {
+                            appendPaddedHTMLComment(file, "  R  " + patch.target().path().get().toString());
+                            appendPaddedHTMLComment(file, "      (" + patch.source().path().get().toString() + ")");
+                        } else if (status.isCopied()) {
+                            appendPaddedHTMLComment(file, "  C  " + patch.target().path().get().toString());
+                            appendPaddedHTMLComment(file, "      (" + patch.source().path().get().toString() + ")");
+                        }
+                    }
+                }
             }
-            Files.writeString(file, "#\n", StandardOpenOption.APPEND);
-            Files.writeString(file, "# Target repository: " + remotePullPath + "\n", StandardOpenOption.APPEND);
-            Files.writeString(file, "# Target branch: " + targetBranch + "\n", StandardOpenOption.APPEND);
+            appendPaddedHTMLComment(file, "");
+            if (issue.isPresent()) {
+                appendPaddedHTMLComment(file, "Issue:      " + issue.get().webUrl());
+            }
+            appendPaddedHTMLComment(file, "Repository: " + parentRepo.webUrl());
+            appendPaddedHTMLComment(file, "Branch:     " + targetBranch);
+
             var success = spawnEditor(repo, file);
             if (!success) {
                 System.err.println("error: editor exited with non-zero status code, aborting");
                 System.exit(1);
             }
             var lines = Files.readAllLines(file)
                              .stream()
-                             .filter(l -> !l.startsWith("#"))
+                             .filter(l -> !(l.startsWith("<!--") && l.endsWith("-->")))
```

Comment:
Perhaps add a .map(String::strip) before this one to prevent whitespace mistakes?

Reply:
Yeah, I was thinking about that, but then I realized that _if_ you want to include an HTML comment in your PR text (like this PR does) then the easiest way to do that is to just insert a space at the end of the sentence :smiley:

Not sure how common it will be to include HTML comments though, this is the first time I've done it, but I think I prefer the "simple" semantics that the line strictly has to start and end with `<!--` and `-->`.